gpii/node_modules/preferencesServer/src/cloudSafeCredHandlers.js

```diff
@@ -0,0 +1,66 @@
+/*!
+GPII Preferences Server Cloud Safe Credentials Handler
+
+Copyright 2014 Raising the Floor - International
+Copyright 2018 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+The research leading to these results has received funding from the European Union's
+Seventh Framework Programme (FP7/2007-2013) under grant agreement no. 289016.
+
+You may obtain a copy of the License at
+https://github.com/GPII/universal/blob/master/LICENSE.txt
+*/
+
+"use strict";
+
+var fluid = require("infusion"),
+    gpii = fluid.registerNamespace("gpii");
+
+fluid.defaults("gpii.preferencesServer.cloudSafeCredCreate.handler", {
+    gradeNames: ["kettle.request.http"],
+    invokers: {
+        handleRequest: {
+            funcName: "gpii.preferencesServer.cloudSafeCredCreate.handler.put",
+            args: ["{gpii.preferencesServer.preferencesService}", "{preferencesServer}", "{that}", "{that}.req.params", "{that}.req.body"]
+        }
+    }
+});
+
+gpii.preferencesServer.cloudSafeCredCreate.handler.put = function (prefsService, prefsServer, request, params, body) {
+    prefsService.cloudSafeCredCreate({
+        prefsSafeId: params.prefsSafeId,
+        username: body.username,
+        password: body.password
+    }).then(request.events.onSuccess.fire, request.events.onError.fire);
+};
+
+fluid.defaults("gpii.preferencesServer.cloudSafeUnlockPost.handler", {
+    gradeNames: ["kettle.request.http"],
+    invokers: {
+        handleRequest: {
+            funcName: "gpii.preferencesServer.cloudSafeUnlockPost.handler.post",
+            args: ["{gpii.preferencesServer.preferencesService}", "{preferencesServer}", "{that}", "{that}.req.params.prefsSafeId", "{that}.req.body"]
+        }
+    }
+});
+
+gpii.preferencesServer.cloudSafeUnlockPost.handler.post = function (prefsService, prefsServer, request, prefsSafeId, body) {
+    prefsService.cloudSafeUnlock({
+        username: body.username,
+        password: body.password
+    }).then(
+        function (data) {
+            request.events.onSuccess.fire(data);
+        },
+        function (/* err */) {
+            request.res.status(401);
```

Comment:
The typical way for line 59-63 is: 
```
        request.events.onError.fire({
            isError: true,
            message: "Unable to unlock a Preferences Safe with the supplied credentials.",
            statusCode: 401
        });
```
The `statusCode` fired with the onError will automatically sets the responded http status code to 401.

Reply:
Thanks, have moved it to our standard idiom!